### PR TITLE
feat: Make it possible to include tags in event envelope

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: akka-persistence-testkit/src/test/scala/akka/persistence/testkit/query/CurrentEventsBySlicesSpec.scala [hunk 1]
```diff
@@ -61,6 +61,24 @@ class CurrentEventsBySlicesSpec
...
+      val ref2 = spawn(testBehaviour("TagTest|pid-2"))
+      ref2 ! Command("evt-3", probe.ref)
+      ref2 ! Command("tag-me-evt-4", probe.ref)
+      probe.receiveMessages(2)
+
+      queries
+        .currentEventsBySlices[String]("TagTest", 0, Persistence(system).numberOfSlices - 1, NoOffset)
+        .runWith(Sink.seq)
+        .futureValue
+        .map(_.tags) should ===(Seq(Some(Set("tag")), Some(Set.empty), Some(Set.empty), Some(Set("tag"))))
```
Comment (patriknw) [inline akka-persistence-testkit/src/test/scala/akka/persistence/testkit/query/CurrentEventsBySlicesSpec.scala:80]: would be good to verify that the actual event is also correct, that it is not Tagged

File: akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala [hunk 2]
```diff
@@ -81,6 +122,8 @@ object EventEnvelope {
  *
  * It is an improved `EventEnvelope` compared to [[akka.persistence.query.EventEnvelope]].
  *
+ * @param tags Some set with tags if journal does support extracting, None if not
```
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:125]: Do we need to distinguish between supporting or not in the api? This envelope is only used by eventsBySlices and we are in control of those implementations. Shouldn't be much harm to use empty Set for "unsupported"?

Comment (johanandren) [reply akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:125]: I thought it'd be good for us to be able to detect not read vs no tags and throw in the filtering logic, not strictly necessary no.

File: akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala [hunk 3]
```diff
@@ -94,7 +133,32 @@ final class EventEnvelope[Event](
     val entityType: String,
     val slice: Int,
     val filtered: Boolean,
-    val source: String) {
+    val source: String,
+    val tags: Set[String]) {
```
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:137]: do we have a serializer that needs to be updated?

Comment (johanandren) [reply akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:137]: Didn't think there was anything else than the gRPC transport but we do have serialisation for it in QuerySerializer, I'll update

Comment (patriknw) [reply akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:137]: I think it's sent over pubsub

File: akka-persistence-query/src/main/scala/akka/persistence/query/internal/QuerySerializer.scala [hunk 4]
```diff
@@ -100,6 +105,7 @@ import akka.serialization.Serializers
 
       val filtered = env.hasFiltered && env.getFiltered
       val source = if (env.hasSource) env.getSource else ""
+      val tags = env.getTagsList.iterator.asScala.toSet
```
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/internal/QuerySerializer.scala:108]: `if (env.hasTags)` or is that not needed for repeated?

Comment (patriknw) [reply akka-persistence-query/src/main/scala/akka/persistence/query/internal/QuerySerializer.scala:108]: might also be good with an isEmpty check to optimize that case and avoid some conversions of empy list

File: akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala [hunk 5]
```diff
@@ -152,6 +216,12 @@ final class EventEnvelope[Event](
     eventMetadata.map(_.asInstanceOf[AnyRef]).asJava
   }
 
+  /**
+   * Java API: empty if the journal does not support extracting tags, non empty optional with tags or empty set
+   * if journal does read tags
```
Comment (patriknw) [inline akka-persistence-query/src/main/scala/akka/persistence/query/typed/EventEnvelope.scala:221]: update docs

File: akka-persistence-query/src/main/scala/akka/persistence/query/internal/QuerySerializer.scala (hunk 4 above)
Comment (johanandren) [reply akka-persistence-query/src/main/scala/akka/persistence/query/internal/QuerySerializer.scala:108]: No hasNnn for repeated, but optimisation makes sense.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**